# 04 — Segment "CLV" Formulas — illustrative only

> **Read first.** Customer Lifetime Value needs repeat purchases per customer over time. This dataset has **no customer
> identifier and no repeat-purchase history**. The three formulas below are applied to the 5 segment labels and are kept
> because the Power BI Profitability page reads `clv_segments.csv`. They demonstrate the formulas; they are **not**
> customer lifetime values and should not be used for financial planning.

Logic: `src/segment_value.py` (`python -m src.segment_value`).

In [1]:
import sys, warnings
from pathlib import Path

ROOT = Path("..").resolve()          # repo root (notebooks live in notebooks/)
sys.path.insert(0, str(ROOT))

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

sns.set_theme(style="whitegrid")
pd.set_option("display.float_format", lambda x: f"{x:,.4f}")

In [2]:
from src.segment_value import clv_by_segment, DISCOUNT_RATE, RETENTION_CLIP

df = pd.read_csv(ROOT / "data/processed/campaign_data_cleaned.csv")
clv = clv_by_segment(df)
clv[["customer_segment", "total_revenue", "total_conversions", "frequency", "AOV",
     "RRetention_Rate", "Retention_Rate", "CLV_simple", "CLV_aov", "CLV_discounted"]]

,customer_segment,total_revenue,total_conversions,frequency,AOV,RRetention_Rate,Retention_Rate,CLV_simple,CLV_aov,CLV_discounted
0,College Students,17243247161,34514710,33533,499.5913,0.9931,0.9500,17243247161,"16,752,793.4336","106,101,025.0793"
1,Premium Shoppers,16893770924,33860455,33071,498.9233,0.9936,0.9500,16893770924,"16,499,893.4074","104,499,324.9138"
2,Tier 2 City Customers,17072782747,34121667,33091,500.3502,0.9950,0.9500,17072782747,"16,557,088.3123","104,861,559.3111"
3,Working Women,17215043588,34482679,33413,499.2374,1.0000,0.9500,17215043588,"16,681,019.8072","105,646,458.7792"
4,Youth,17225401651,34533588,33557,498.8014,0.9917,0.9500,17225401651,"16,738,278.2004","106,009,095.2693"


## Formulas and assumptions
| Column | Formula | Assumption / problem |
|---|---|---|
| `AOV` | total revenue / total conversions | valid: average revenue per conversion (≈ 499 in the data's currency) |
| `CLV_simple` | total segment revenue | this is just revenue, not a lifetime value |
| `CLV_aov` | AOV × number of campaigns | multiplies a per-order value by a campaign count — mixed grain |
| `Retention_Rate` | avg engagement / max avg engagement, clipped to [0.10, 0.95] | **proxy** — engagement is not retention |
| `CLV_discounted` | AOV × count × r / (1 + d − r), d = 10% | standard margin-multiple form, but every input above is a proxy |

In [3]:
print("discount rate:", DISCOUNT_RATE, "| retention clip:", RETENTION_CLIP)
print("unclipped retention proxy range:", clv.RRetention_Rate.min().round(4), "to", clv.RRetention_Rate.max().round(4))
print("CLV_simple == total_revenue for every segment:", bool((clv.CLV_simple == clv.total_revenue).all()))
spread = clv[["AOV", "CLV_discounted"]].agg(lambda s: s.max() / s.min() - 1)
print("max vs min segment spread:", (spread * 100).round(2).astype(str).add("%").to_dict())

discount rate: 0.1 | retention clip: (0.1, 0.95)
unclipped retention proxy range: 0.9917 to 1.0
CLV_simple == total_revenue for every segment: True
max vs min segment spread: {'AOV': '0.31%', 'CLV_discounted': '1.53%'}


## Sensitivity: the result is driven by the assumed retention, not by the data

In [4]:
r = np.array([0.50, 0.70, 0.80, 0.90, 0.95])
base = clv.loc[0, "AOV"] * clv.loc[0, "frequency"]
sens = pd.DataFrame({"assumed_retention": r, "CLV_discounted": base * r / (1 + DISCOUNT_RATE - r)})
sens["vs_r_0.95"] = sens["CLV_discounted"] / sens["CLV_discounted"].iloc[-1]
sens

,assumed_retention,CLV_discounted,vs_r_0.95
0,0.5000,"13,960,661.1946",0.1316
1,0.7000,"29,317,388.5088",0.2763
2,0.8000,"44,674,115.8229",0.4211
3,0.9000,"75,387,570.4511",0.7105
4,0.9500,"106,101,025.0793",1.0000


## Interpretation
* AOV (~499) is the one sound number here and it is nearly identical across segments.
* Because the retention proxy is clipped to 0.95 for all five segments, `CLV_discounted` differs between segments only
  through AOV × campaign count, i.e. by about 1–2%.
* The sensitivity table shows the "CLV" moves several-fold with the retention assumption, which the data cannot supply.
* **Conclusion:** no segment should be prioritised on these CLV values. A real CLV needs customer-level orders
  (e.g. BG/NBD + Gamma-Gamma, or cohort retention curves).